src/go/rpk/pkg/cli/cmd/redpanda/start_test.go

```diff
@@ -1680,3 +1681,143 @@ func TestExtraFlags(t *testing.T) {
 		})
 	}
 }
+
+func Test_buildRedpandaFlags(t *testing.T) {
+	type args struct {
+		conf       *config.Config
+		args       []string
+		sFlags     seastarFlags
+		flags      map[string]string
+		ioResolver func(*config.Config, bool) (*iotune.IoProperties, error)
+	}
+	tests := []struct {
+		name       string
+		args       args
+		writeIoCfg bool
+		exp        map[string]string
+		expErr     bool
+	}{
+		{
+			name: "err when ioPropertiesFlag and wellKnownIo are set",
+			args: args{
+				conf:  &config.Config{Rpk: config.RpkConfig{WellKnownIo: "some io"}},
+				flags: map[string]string{ioPropertiesFlag: "{some:value}"},
+			},
+			expErr: true,
+		}, {
+			name: "err when ioPropertiesFileFlag and wellKnownIo are set",
+			args: args{
+				conf:  &config.Config{Rpk: config.RpkConfig{WellKnownIo: "some io"}},
+				flags: map[string]string{ioPropertiesFileFlag: ""},
+			},
+			expErr: true,
+		}, {
+			name: "setting the properties from the config file ",
+			args: args{
+				conf: &config.Config{Rpk: config.RpkConfig{
+					Overprovisioned:      true,
+					EnableMemoryLocking:  false,
+					SMP:                  intPtr(2),
+					AdditionalStartFlags: []string{"--abort-on-seastar-bad-alloc=true"},
+				}},
+			},
+			exp: map[string]string{
+				"overprovisioned":            "true",
+				"lock-memory":                "false",
+				"smp":                        "2",
+				"abort-on-seastar-bad-alloc": "true",
+			},
+		}, {
+			name: "err if flag and additional_start_flags are present",
+			args: args{
+				flags: map[string]string{maxIoRequestsFlag: "2"},
+				conf: &config.Config{Rpk: config.RpkConfig{
+					AdditionalStartFlags: []string{"--max-io-requests=3"},
+				}},
+			},
+			expErr: true,
+		}, {
+			name: "get the io property file from the default location",
+			args: args{
+				conf: &config.Config{},
+			},
+			writeIoCfg: true,
+			exp: map[string]string{
+				"io-properties-file": "io-config.yaml",
+				// These 2 are false because of the empty config
+				"overprovisioned": "false",
+				"lock-memory":     "false",
+			},
+		}, {
+			name: "get the io property from the  ioResolver",
+			args: args{
+				conf: &config.Config{},
+				ioResolver: func(_ *config.Config, _ bool) (*iotune.IoProperties, error) {
+					return &iotune.IoProperties{
+						MountPoint:     "/mnt/",
+						ReadIops:       2,
+						ReadBandwidth:  3,
+						WriteIops:      4,
+						WriteBandwidth: 5,
+					}, nil
+				},
+			},
+			exp: map[string]string{
+				"io-properties": `'disks:
+    - mountpoint: /mnt/
+      read_iops: 2
+      read_bandwidth: 3
+      write_iops: 4
+      write_bandwidth: 5
+'`,
```

Comment:
This will change to JSON once we merge https://github.com/redpanda-data/redpanda/pull/8298

Reply:
Changed :+1: